packages/itwin/grouping-mapping-widget/src/widget/components/Grouping.tsx

```diff
@@ -390,47 +402,91 @@ export const Groupings = ({ mapping, goBack }: GroupsTreeProps) => {
               await goBack();
             }}
           />
-          <Surface className="gmw-groups-container">
-            <div className="gmw-toolbar">
-              <Button
-                className="add-load-button"
-                startIcon={
-                  isLoadingQuery ? (
-                    <ProgressRadial size="small" indeterminate />
-                  ) : (
-                    <SvgAdd />
-                  )
-                }
-                styleType="high-visibility"
+
+          <Surface className='gmw-groups-container'>
+            <div className='gmw-toolbar'>
+              <DropdownMenu
+                className='gmw-ui-provider-dropdown'
                 disabled={isLoadingQuery}
-                onClick={addGroup}
+                menuItems={() =>
+                  uiProviders.length > 0
+                    ? uiProviders.map((p) => (
+                      <MenuItem
```

Comment:
Please reduce code duplication. You could do some kind of concatenation to render all `MenuItem`s in a single `.map`.